instrumentation/cats-effect/cats-effect-common-3.6/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/catseffect/common/v3_6/IoLocalContextSingleton.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.catseffect.common.v3_6;
+
+import application.io.opentelemetry.context.Context;
+import cats.effect.IOLocal;
+import io.opentelemetry.javaagent.instrumentation.opentelemetryapi.context.AgentContextStorage;
+
+public class IoLocalContextSingleton {
```

Comment:
It must be defined in a `common` library so we can lately reuse it to instrument [otel4s](https://github.com/typelevel/otel4s).